homeassistant/components/alexa_local_control.py

```diff
@@ -0,0 +1,527 @@
+    EVENT_HOMEASSISTANT_START
+)
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS, ATTR_SUPPORTED_FEATURES, SUPPORT_BRIGHTNESS
+)
+
+DOMAIN = "alexa_local_control"
+
+REQUIREMENTS = ['Flask==0.11.1']
+
+_LOGGER = logging.getLogger(__name__)
+
+# Note that the double newline at the end of
+# this string is required per the SSDP spec
+SSDP_RESPONSE = """HTTP/1.1 200 OK
+CACHE-CONTROL: max-age=60
+EXT:
+LOCATION: http://{0}:{1}/description.xml
+SERVER: FreeRTOS/6.0.5, UPnP/1.0, IpBridge/0.1
+ST: urn:schemas-upnp-org:device:basic:1
+USN: uuid:Socket-1_0-221438K0100073::urn:schemas-upnp-org:device:basic:1
+
+"""
+
+DESCRIPTION_XML_RESPONSE = """<?xml version="1.0" encoding="UTF-8" ?>
+<root xmlns="urn:schemas-upnp-org:device-1-0">
+<specVersion>
+<major>1</major>
+<minor>0</minor>
+</specVersion>
+<URLBase>http://{0}:{1}/</URLBase>
+<device>
+<deviceType>urn:schemas-upnp-org:device:Basic:1</deviceType>
+<friendlyName>HA-Echo ({0})</friendlyName>
+<manufacturer>Royal Philips Electronics</manufacturer>
+<manufacturerURL>http://www.philips.com</manufacturerURL>
+<modelDescription>Philips hue Personal Wireless Lighting</modelDescription>
+<modelName>Philips hue bridge 2015</modelName>
+<modelNumber>BSB002</modelNumber>
+<modelURL>http://www.meethue.com</modelURL>
+<serialNumber>1234</serialNumber>
+<UDN>uuid:2f402f80-da50-11e1-9b23-001788255acc</UDN>
+<presentationURL>index.html</presentationURL>
+<iconList>
+<icon>
+<mimetype>image/png</mimetype>
+<height>48</height>
+<width>48</width>
+<depth>24</depth>
+<url>hue_logo_0.png</url>
+</icon>
+<icon>
+<mimetype>image/png</mimetype>
+<height>120</height>
+<width>120</width>
+<depth>24</depth>
+<url>hue_logo_3.png</url>
+</icon>
+</iconList>
+</device>
+</root>
+"""
+
+CONF_HOST_IP = 'host_ip'
+CONF_LISTEN_PORT = 'listen_port'
+CONF_OFF_MAPS_TO_ON_DOMAINS = 'off_maps_to_on_domains'
+CONF_EXPOSE_BY_DEFAULT = 'expose_by_default'
+CONF_EXPOSED_DOMAINS = 'exposed_domains'
+
+ATTR_ECHO = 'echo'
+ATTR_ECHO_FRIENDLY_NAME = 'echo_friendly_name'
+
+DEFAULT_LISTEN_PORT = 8300
+DEFAULT_OFF_MAPS_TO_ON_DOMAINS = ['script', 'scene']
+DEFAULT_EXPOSE_BY_DEFAULT = True
+DEFAULT_EXPOSED_DOMAINS = [
+    'switch', 'light', 'script', 'scene', 'group', 'input_boolean',
+    'media_player'
+]
+
+
+def setup(hass, config):
+    """Activate the alexa_local_control component."""
+    from flask import Flask
+
+    app = Flask(__name__)
+    app.url_map.strict_slashes = False
+
+    view = HueBridgeView(hass, config)
+    view.set_up_routes(app)
+
+    def start_listener(event):
+        """Start listening for UPNP/SSDP requests."""
+        upnp_listener = UPNPResponderThread(
+            view.host_ip_addr, view.listen_port)
+        upnp_listener.start()
+
+    def start_web_server():
+        """Start the Hue bridge API web server."""
+        from cherrypy import wsgiserver
+
+        dispatcher = wsgiserver.WSGIPathInfoDispatcher({'/': app})
+        server = wsgiserver.CherryPyWSGIServer(

```

Comment:
Are you creating a new webserver/using Flask because Hue needs to run on it's own port? I believe that we can use any port as long as we announce it, no? See previous comment questioning use of Flask.